### ISSUE TITLE
Collections::PermissionService.access_to_collection? is using a PermissionTemplate to decide access.

### ISSUE THREAD

**Issue by jcoyne** [samvera/hyrax#2474]
A `PermissionTemplate` is just a template for creating access controls when an object is first created.  It should not be consulted to determine access controls on an existing collection.  `PermissionTemplate`s are intended to be ephemeral, and thus they should be able to be destroyed without affecting the access to any object in the repository. All access controls should be stored in the repository (Fedora) using WebACL.

**Comment by elrayle** [comment]
@jcoyne Can you be more specific?  I do not believe that we have done anything inconsistent with existing code at master.  

https://github.com/samvera/hyrax/blob/master/app/models/concerns/hyrax/ability.rb#L57-L84 which queries the PermissionTemplateAccess table to determine admin sets for management and deposit.

This is a similar query to those in Collections::PermissionService. 

Ultimately in both master and collections-sprint, a user cannot gain access to a collection or admin set unless they have the appropriate edit/read access in a Fedora WebACL.

https://github.com/samvera/hyrax/blob/collections-sprint/app/models/concerns/hyrax/ability/collection_ability.rb#L15-L17
https://github.com/samvera/hyrax/blob/collections-sprint/app/models/concerns/hyrax/ability/collection_ability.rb#L33-L35

**Comment by jcoyne** [comment]
I agree that there is a bug in the current behavior as well.  We should fix it and not perpetuate it.  We should not look at a Template to determine access for existing objects.

**Comment by jcoyne** [comment]
It looks like we could use `Control` for the manage access.  https://www.w3.org/wiki/WebAccessControl#Modes_of_Access we may need to create our own predicate for `deposit` unless you think `Append` would be sufficient?  When you deposit a work in a collection, do you have the rights to remove it from the collection, or is that permanent?

**Comment by elrayle** [comment]
@jcoyne I agree with your assessment that it would be better to have deposit access part of the access control system in Fedora/Solr.  This would allow for the ability checks to call `test_deposit(id)` in much the same way they currently use `test_read(id)` and `test_edit(id)`.  This is a major refactor of existing code and CE code and is beyond the scope of the CE work.  If however, you wish to take this on, I would be in support of that work.

RE: Can deposit users remove a collection/work they deposited into a collection?  At this point they cannot remove works/collections from a collection.  This is not optimal.  As a user, I could reasonably expect to add a work and immediately decide it shouldn't be there.  It is arbitrary from the user's perspective that they cannot remove a work they added.

Why is this the case?  There is not a way to tell which collections/works the current user added to the collection.  This leaves us with the options that the depositor can remove none (as it currently is) or any that the user can see regardless of whether the user added the collection/work or not.  

For our local use cases, I can see going with the second option.  Or perhaps even better is to make this configurable and let sites decide.

**Comment by cjcolvar** [comment]
I think this needs some design work that might be influenced by https://wiki.duraspace.org/display/samvera/Samvera+Permissioning+Analysis+and+Design+Working+Group .  This design work, the choosing of predicates, and the data migration required to push the values in the DB to WebACLs are probably the bulk of the work.  My assumption is that once those are done the rest should follow pretty smoothly.

**Comment by elrayle** [comment]
Thanks @cjcolvar for doing an initial analysis of what would need to be done to address this issue.

This is a significant refactor and will not be done in the Hyrax 2.1 timeframe.

**Comment by elrayle** [comment]
I am removing the blocker tag for 2.1.0.  This is targeted for 2.1.1.
